src/nupic/algorithms/Cells4.cpp

```diff
@@ -1628,46 +1667,55 @@ void Cells4::adaptSegment(const SegmentUpdate& update)
     // updateSynapses? It would be cleaner and avoid the need to keep the
     // synapses sorted.
 
-    // Accumulate list of synapses to decrement, increment, add, and remove
-    std::set<UInt> synapsesSet(update.begin(), update.end());
+    // Accumulate lists of synapses to decrement, increment, add, and remove
+    std::set<UInt> synapsesSet(update.begin(), update.end()); // srcCellIdx
```

Comment:
@scottpurdy: I followed an existing example. Since they are declared as set or vectors of UInt, and the variables are named poorly (IMHO), it's difficult for the reader/maintainer of the code to know what sort of animal is stored in these containers. Some of them store indexes of synapses within the segment; others store source cell indexes. So, srcCellIdx means that it contains source cell indexes - I borrowed it from this existing code: https://github.com/numenta/nupic.core/pull/1236/files/1bdacbd7cac66dd44c342a7bb8fd198fa423fb29#diff-4c232a262b217bd7ab4e516856187881R1672